### DISCUSSION TITLE
Debugging HTTP promotion steps

### DISCUSSION THREAD

**Discussion by muenchdo** [akuity/kargo#3084] [General]
Congrats on the launch of v1.1.0! 🎉 Both HTTP promotion steps and the expression language are major features I've been looking forward to.

Do you already have any guidance on how to troubleshoot HTTP promotion steps? In my case, I'm receiving an error when parsing the response (see below), but I haven't found a way to actually see the response body. I increased the log level to debug on the controller, but that didn't give me any extra info.

> step execution failed: step 0 met error threshold of 1: failed to run step "http": error extracting outputs from HTTP response: error evaluating expression: index out of range: 0 (array length is 0) (1:21) | response.body.issues[0].key | ....................^

**Comment by krancour** [comment]
> Congrats on the launch of v1.1.0! 🎉 Both HTTP promotion steps and the expression language are major features I've been looking forward to.

Thank you for the warm feedback!

Since it seems the response to the request isn't what you were expecting, my best recommendation is to use curl locally to emulate the request and see how it differs from what you were expecting...

But I assume you'd have tried that already...

If you want to see what what Kargo's seeing, _and provided you don't expect anything sensitive in the response_, you could try writing all of `response.body` to the output and then have a look at the Promotion's status to see what you got.
